### DISCUSSION TITLE
Support multi database

### DISCUSSION THREAD

**Discussion by selvam920** [simolus3/drift#1246] [General]
i have the below requirement.
-> 1 master database
-> multiple detail database
when login the app master database will use, after login any one detail database will use.
how to do connection with one detail database after initialize app?
Pls suggest, becoz every time i'm initializing the database

**Comment by simolus3** [comment]
You can use multiple moor databases in the your app, which is what I would do here. It's hard to give exact advice without additional details, but you could have one moor database class for the master database and (potentially more than) one for the detail databases. Initially you'd use the master database, when desired you can just switch over.

How to switch between database classes is a concern of your app, but moor lets you use multiple databases in the same app.

**Comment by selvam920** [comment]
How to create instance for the one detail database?

**Comment by selvam920** [comment]
Currently I'm creating instance in static class and using everywhere in the app

**Reply by simolus3** [reply]
That should work. Depending on your app's architecture there could be other patterns like `get_it` or `provider`s that may be helpful. [This page](https://moor.simonbinder.eu/faq/#using-the-database) lists some options.

**Reply by selvam920** [reply]
Using sqlite3 dll in windows desktop app, 
Vscode debug is getting stuck every time. 
That's why I asked this question to create instance in static class

**Reply by simolus3** [reply]
Do you get an exception? If so, can you please post it?

**Reply by selvam920** [reply]
No, every initial time getting stuck for 2 min after the debug is working fine

**Reply by simolus3** [reply]
I'll need more information to diagnose this. Where exactly is it getting stuck? Can you try to create a small project that reproduces this for me to take a look at?

**Comment by selvam920** [comment]
Is this good way?
